### PR TITLE
types: add carbon__elements

### PR DESCRIPTION
**Pull request by metonym**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/carbon__elements/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6458 @@
+// Type definitions for @carbon/elements 10.24
+// Project: https://github.com/carbon-design-system/carbon/tree/master/packages/elements
+// Definitions by: Eric Liu <https://github.com/metonym>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.1
```
Comment (peterblazejewicz) [inline types/carbon__elements/index.d.ts:5]: this is month old release,
is there a specific reason to use 4.1 only?
https://github.com/microsoft/TypeScript/releases/tag/v4.1.2

Comment (metonym) [reply types/carbon__elements/index.d.ts:5]: No reason in particular. Should I downgrade to 4.0?

Comment (peterblazejewicz) [reply types/carbon__elements/index.d.ts:5]: @metonym not sure, other @carbon/* stuff seems to have 4.0, 3.8 or no versioning at all.
Let's keep it 'as is' for now. Usually that versioning is due to specific language features inroduced
with e.g. 4.1 version (like named tuples, or something similar).

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Pull request merged**